File: 14-08/buscaElementoMatriz.py


# Receber numero de linhas e colunas da matriz
nLinhas = int(input("Digite o numero de linhas da matriz: "))
nColunas = int(input("Digite o numero de colunas da matriz: "))

# Declarar a matriz
matriz = []

# Entrada de dados da matriz
for i in range(nLinhas):
    linha = []
    for j in range(nColunas):
        elemento = int(input(f"Digite o elemento [{i+1}] [{j+1}] da matriz: "))
        linha.append(elemento)
    matriz.append(linha)

# Elemento que deve ser procurado
elemento_procurado = int(input("Digite o elemento que deve ser procurado: "))

# Ver se o elemento está presente na matriz
encontrado = False
for i in range(nLinhas):
    for j in range(nColunas):
        if matriz[i][j] == elemento_procurado:
            print(f"Elemento está presente na matriz, linha {i+1} coluna {j+1}")
            encontrado = True

# Caso o elemento não esteja na matriz
if not encontrado:
    print("Elemento não está presente na matriz")

File: 14-08/buscaElementoMatriz.c


#include <stdio.h>

int main(){
    // Declaração de variaveis
	int elemento, nLinhas, nColunas, encontrado;
    
    // Receber numero de linhas e colunas da matriz
	printf("Digite o numero de linhas da matriz: ");
	scanf("%d", &nLinhas);

	printf("Digite o numero de colunas da matriz: ");
	scanf("%d", &nColunas);
    
    // Declarar a matriz
	int matriz[nLinhas][nColunas];
    
    // Entrada de dados da matriz
	for(int i = 0; i < nLinhas; i++){
		for(int j = 0; j < nColunas; j++){
			printf("Digite o elemento [%d] [%d] da matriz: ", i, j);
			scanf("%d", &matriz[i][j]); 
		}
	}
    
    // Elemento que deve ser procurado
	printf("Digite o elemento que deve ser procurado: ");
	scanf("%d", &elemento);
	
	// Ver se elemento esta presente na matriz
	for(int i = 0; i < nLinhas; i++){
		for(int j = 0; j < nColunas; j++){
			if(elemento == matriz[i][j]){
				printf("Elemento está presente na matriz, linha %d coluna %d\n", i+1, j+1);
			    encontrado = 1;
			}
			
		}
	}
    // Caso elemento não esteja na matriz
    if(!encontrado){
        printf("Elemento não está presente na matriz");
    }
	return 0;
}

